contracts/Removal.sol

```diff
@@ -80,37 +110,87 @@ contract Removal is
     return removalId.unpackRemovalIdV0();
   }
 
+  // TODO do we want a batch version of this?
+  /**
+   * @notice Get the restriction schedule id (which is the removal's project id) for a given removal id.
+   */
+  function getProjectIdForRemoval(uint256 removalId)
+    public
+    view
+    returns (uint256)
+  {
+    return _removalIdToProjectId[removalId];
```

Comment:
I find the DX really strange here since we get removal information for a given removalId otherwise from RemovalUtils

Reply:
were there any other alternatives you explored?

Reply:
I think it's just that we're out of space on the id itself to store more information in it, so now that a removal maps to the notion of a project id we just have to store it outside the removal id

Reply:
is there nothing we could drop from the removal id that we don't want to use?

Reply:
the only thing that comes to mind would be the parcel identifier... I think we have a few bytes to work with there.  since the project id is used on-chain and the parcel identifier isn't (yet) (and seems less likely to be important) perhaps we could replace that field with the project id instead and if we ever need to know which parcel a removal is associated with (such as for rendering the front-end "gallery" view of sellers likely to comprise an order) we can just look that up with database queries for the removal's parcel parent?  this might actually be worth doing.... would definitely want to do so in a separate ticket/PR so as not to muddy this giant one but could be a worthy refactor.

ACTUALLY, the field on the removal id we use for parcel id is just called "`subIdentifier`" ... we could repurpose that however we want with pretty minimal changes.  maybe just changing its name to be more descriptive and then pulling the project id from that field.  we already have a bunch of overhauling to do in the app code for minting and listing so we would take care of necessary changes there.

Reply:
parcel id + vintage are the only sources of uniqueness in the removal id space for a given project so I don't think there's a path to removing that.

I think it would make sense to prioritize the space in the removal id for what's needed or anticipated to be needed in the purchase flow vs data we might want to paint a complete picture of the project in registry since the former is likely to be used in gas paying transactions and the later only in view functions.

Reply:
ah ok... I just pushed a commit (that I am happy to undo) that uses that space on the removal id as a project id instead of as a parcel id.  I'm wondering if we really do feel like we need the parcel identifier on the removal itself since this is 1) meaningless outside of the context of our datastore anyway and 2) a connection that exists between removal id and parcel id in our datastore off-chain even if the id doesn't encode the parcel id directly.  We don't yet have a use case for taking advantage of the parcel id on chain (and it seems confusing to me that we would ever need to filter orders at that level of granularity such that buyers care about which specific field they're buying from over something simpler like state or country), whereas we do have a clear need for the slightly less granular notion of a project id and can definitely take advantage of gas savings by encoding that directly.
What do you think?  Still keep the parcel id?

Reply:
ohhhhh, yep, I get it now.  The ids themselves won't be unique enough without being comprised by those fields. bummer

Reply:
This is starting to make me think having the supplier address encoded in the ID is the thing that should be dropped instead-- especially since we've already identified a potential problem with relying on it in rNORI. Happy to punt this conversation into a ticket/todo if that's not a clear cut path forward